### ISSUE TITLE
Provide a ‘Skip to Content’ link

### ISSUE THREAD

**Issue by Angamanga** [ushahidi/platform#2746]
## Overview
Screen readers announce all of the links on the page to the user, in order. To help navigation, provide a way for them to bypass parts of each page, for example such as long navigation links in mode-bar, filter-bar in the toolbar and each map-icon.

## Requirements
- Add a ‘Skip to Content’ link to all views that skips the mode-context-bar, filter-bar and goes direct to the "main-window"
https://www.wuhcag.com/bypass-blocks/

## Acceptance criteria
- You should be able to use tab to get to the 'skip to content' as the second or third 'tab' (as in the keyboard button tab) the first should be content/title that orientates the non-sighted user as to what is on the whole site. In our case I would suggest it go:

1 - Deployment name
2 - Deployment description
3 - Skip to content (we need to define what we consider 'content' here I would suggest that we define content as 'posts' an go right to the pins/posts)

Aha! Link: https://ushahiditeam.aha.io/features/PROD-372

**Comment by rowasc** [comment]
@Angamanga @sethburtonhall is this in the correct pipeline ? (Ready for Dev)

**Comment by Angamanga** [comment]
@rowasc Moving to ready for spec

**Comment by Erioldoesdesign** [comment]
@Angamanga Does this require any design work? I'm not sure it does but worth checking!

**Comment by willdoran** [comment]
@sethburtonhall  @Angamanga I'm stuck on the acceptance criteria section, how do we test this work?

**Comment by Erioldoesdesign** [comment]
@willdoran (not sure this is helpful but) you should be able to use tab to get to the 'skip to content' as the second or third 'tab' the first should be content/title that orientates the non-sighted user as to what is on the whole site. In our case I would suggest it go:

1 - Deployment name
2 - Deployment description
3 - Skip to content (we need to define what we consider 'content' here I would suggest that we define content as 'posts' an go right to the pins/posts)

Keen to get other folks input though @justinscherer 

Also I can ask some of the inclusivity community around these parts 👍

**Comment by willdoran** [comment]
@Erioldoesdesign Thanks that's very helpful I'll include that as the acceptance criteria and request reviews from @justinscherer @Angamanga and yourself.

**Comment by rjmackay** [comment]
@Erioldoesdesign @Angamanga should the skip to content link be visible to all users? or similar to the github approach, only visible if you tab to it?

**Comment by Angamanga** [comment]
@willdoran @Erioldoesdesign @rjmackay When I first wrote this issue, I was thinking about screen-readers only and hiding the skip-to-content link to everybody else. But it actually makes sense to keep it visible for all users since tabbing around everywhere is annoying. But that said, since we don't manage tabbing in the "old" ui right now, I think it makes more sense just keeping it for the screen-reader-users and hiding it for everyone else = no design needed. For the ui-refresh, we should include a skip-link for everybody. 

Suggestion: We'll focus on screen readers and hide the link in this cycle and add I will add an issue for design in the ui-refresh-epic? We could also use something similar to this: http://paypal.github.io/skipto/

**Comment by willdoran** [comment]
@Angamanga Sounds like a great approach, can you update the spec with corrections to what I made?

**Comment by Erioldoesdesign** [comment]
@rjmackay I don't think it's very useful in our case for all users. Visible only if tabbed would be great

**Comment by rowasc** [comment]
@Angamanga implementation for this should be a simple anchor right? Thinking it might be a good code-in issue

**Comment by Angamanga** [comment]
@rowasc Yes, it would be a good code-in issue, more information about it with recommendations can be found here: https://www.w3.org/TR/WCAG20-TECHS/G1.html But I can add a more detailed spec for it as well.

**Comment by bimbolabuari** [comment]
Hi @Angamanga, I'm an outreachy internship applicant. Please can I contribute to this issue?

**Comment by onyeepeace** [comment]
Hi @Angamanga , I am interested in fixing this. Can you please assign this to me? I will also write out my procedure on working on this as I contribute and link it as well.

**Comment by Angamanga** [comment]
@bimbolabuari and @onyeepeace Thank you for your interest in contributing! Please go ahead! 🎉

**Comment by bimbolabuari** [comment]
Thanks @Angamanga, working on it

**Comment by onyeepeace** [comment]
Okay, great. Thanks @Angamanga

**Comment by onyeepeace** [comment]
Hi @Angamanga, I will like to ask a quick question. Should this be implemented on the platform repository or the platform-client repository? I am a little confused on the two repositories and where to start on. Thanks.

**Comment by Angamanga** [comment]
@onyeepeace The platform-client repository. Check out our docs for the setup:  https://docs.ushahidi.com/platform-developer-documentation/development-and-code/setup_alternatives#platform-api-backend-install-guides and here: https://docs.ushahidi.com/platform-developer-documentation/development-and-code/setup_alternatives#platform-api-backend-install-guides. You need to setup both the backend and the client in order to run it locally.

**Comment by onyeepeace** [comment]
Thank you so much @Angamanga. I will set it up now.

**Comment by onyeepeace** [comment]
I am currently trying to set up the dev-environment locally on my computer, I have gone through a lot of steps and I am at the point where I need to run 'vagrant up'. I did that but I am getting this error;
C:\Users\Peace\Desktop\platform>vagrant up
Check your Homestead.yaml file, the path to your private key does not exist.

I found possible solutions here https://stackoverflow.com/questions/44463987/homestead-installation. I am a little confused as to which one to use as I don't want to spoil what I have done previously. Please I need some help @Angamanga

**Comment by Angamanga** [comment]
@onyeepeace Make sure you are in the platform-api folder before running vagrant up, its in this folder the Homestead.yaml file is.

**Comment by rowasc** [comment]
hey @Angamanga who is assigned to this one? It's not clear and I'm worried maybe there's two folks working on it at the same time? @bimbolabuari I'm assuming you are working on this one no?

**Comment by bimbolabuari** [comment]
Hi @rowasc, Yes, I am working on it. Thanks for assigning it to me.

**Comment by bimbolabuari** [comment]
Hi @rowasc and @Angamanga, Please I need your help on where to implement the "skip to content" link, as it is not clear since the root page (http://localhost:8001/views/map) and other links I have access to doesn't seem to have main content

Thanks.

**Comment by rowasc** [comment]
@bimbolabuari in the case of the map, it's also unclear to me how we define content. I *assume* it's the posts in the map. In the  data view however, "skip to content" should "take" the user directly to listen/read the first post card on the left. 
You can start with the data view, and we can all continue the discussion on this ticket on monday @bimbolabuari since I definitely need @Angamanga to help define it.

**Comment by Angamanga** [comment]
@rowasc @bimbolabuari Lets start with the data-view, that is a huge win and once we have decided how to handle the map we can add it to that view :).

**Comment by bimbolabuari** [comment]
@rowasc @Angamanga Thanks for the clarification, I will start with the data-view however there is no seed data for the post view, and am having an issue creating a new post so as to get the idea of skipping to the main content for the view. Please is there a way I can create a new post from the backend?
I am having this error on creating a new post:
```
FUNCTION ushahidi.GeomFromText does not exist [ INSERT INTO `post_point` (`value`, `form_attribute_id`, `post_id`, `created`) VALUES (GeomFromText('POINT(32.958984 -8.928487)'), 1, 12, 1602502299) ]
```
![data-view-post-err](https://user-images.githubusercontent.com/66112760/95741462-363c3b00-0c86-11eb-9dad-7ec22493b1c5.png)

![skip-to-content-err](https://user-images.githubusercontent.com/66112760/95741481-3b00ef00-0c86-11eb-80bf-bfc2b874ca23.png)

**Comment by bimbolabuari** [comment]
@rowasc I am using vagrant set up for backend.

**Comment by bimbolabuari** [comment]
pinging @Angamanga @rowasc

**Comment by Angamanga** [comment]
@rowasc Could you please check?

**Comment by rowasc** [comment]
@bimbolabuari that's odd, I think I can link this to a mysql 8 thing but... I don't see why you'd be in mysql8 if you're using our vagrant box. 
Can you check `mysql --version` inside the vagrant box? 
Thanks

**Comment by bimbolabuari** [comment]
@rowasc checking `mysql --version` shows that the vagrant box is using mysql 8. 


![image](https://user-images.githubusercontent.com/66112760/96337459-3d48bc00-107f-11eb-96f3-39c2d23dc6e1.png)

**Comment by rowasc** [comment]
ah , that's odd. Are you using the same vagrant box we are using, or the new/standard homestead box?
I have `mysql  Ver 14.14 Distrib 5.7.29, for Linux (x86_64)` this in mine.

**Comment by bimbolabuari** [comment]
Not sure if I am using the same vagrant box. However, I followed the setup instruction in the documentation
https://docs.ushahidi.com/platform-developer-documentation/development-and-code/setup_alternatives/vagrant-setup-
Currently using vagrant version 2.2.10 and also virtual box version 6.1.14 
Do you suggest I downgrade the vagrant version?

**Comment by rowasc** [comment]
@bimbolabuari no you should not need to change the version of vagrant itself. I'm confused since that version is even lower than the one I have. You may be able to make the change in Homestead.yml to avoid mysql8
https://stackoverflow.com/questions/61189727/laravel-homestead-downgrade-to-mysql-5-7 similar to this
```
features:
    - mysql8: false
```` 
And if that fails I guess uninstalling mysql8 in the box itself and reinstalling manually will do the trick (not ideal ..  I definitely need to look more into why the setup is defaulting to mysql8 now)
LMK if the Homestead.yml change above works after re provisioning and we can submit a PR to change it for everyone if that's the case.
Thanks

**Comment by bimbolabuari** [comment]
@rowasc, I made the changes in homestead.yml 

```yml
features:
    - mysql8: false
```
and also remove the vagrant box and re-provisioned it but still unable to create a post.
mysql version is still version 8. I tried the second option by uninstalling mysql8 but ubuntu0.20.04.4 is only compatible with mysql8.

**Comment by bimbolabuari** [comment]
Good morning @rowasc, I eventually get my mysql Ver 14.14 Distrib 5.7.31, for Linux (x86_64) after some search
![Screenshot from 2020-10-22 08-16-05](https://user-images.githubusercontent.com/66112760/96837953-19022c00-143f-11eb-9239-e45d18c025fa.png)

I want to be sure it works, will get back to you

Thank you!

**Comment by bimbolabuari** [comment]
@rowasc, I have been able to create a new post!

I discovered that the default version for mysql is version 8 for the latest version of laravel homestead box [here](https://app.vagrantup.com/laravel/boxes/homestead/versions/10.0.0). So i changed the version to `9.6.1` in the code below:

```rb     
 # vendor/laravel/homestead/scripts/homestead.rb
 config.vm.box_version = settings["version"] ||= "9.6.1"
```

This adds ubuntu 18 and mysql 5.7 (`mysql  Ver 14.14 Distrib 5.7.31, for Linux (x86_64)`)
 
Thank you @rowasc!

Can I open a pull request to add this to the documentation?

**Comment by Erioldoesdesign** [comment]
Hi there! sorry I'm stepping in late! I noticed some comments on how skip to content on map should be handled and I'm wondering how important it is to have the skip to content read out the about information about the deployment first (the information that sits under the logo circle)

For sure, on a first users visit the info blurb first to better understand what the deployment is doing to give context to the posts. I wonder if we can cookie know that user has visited recently and skip reading out the blurb...hmm

For data view...this is a bit of an odd journey right? unless the link that a user clicks takes them straight to the deployment data view (it defaults to map right?) then the user needs to get through the nav items to select data view to then get to this skip to content link.

Once on the data view page what's being done is perfect, there should be a way to get into the first post in the list, open, then read through the info but there should also be an indication of how much data is in a single post so perhaps the screen readers need to read out the fields included in the first post and confirm they want to read and then move on to second post.

I'm happy to either like, chat through or perhaps do some prototypes/documents of the 'flow' of this but super happy this is being started 😄
